src/rpc/core_rpc_server.cpp

```diff
@@ -3570,6 +3570,93 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_get_height_by_time(const COMMAND_RPC_GET_HEIGHT_BY_TIME::request& req, COMMAND_RPC_GET_HEIGHT_BY_TIME::response& res, epee::json_rpc::error& error_resp, const connection_context *ctx)
+  {
+    RPC_TRACKER(get_height_by_time);
+    bool r;
+    if (use_bootstrap_daemon_if_necessary<COMMAND_RPC_GET_HEIGHT_BY_TIME>(invoke_http_mode::JON_RPC, "get_height_by_time", req, res, r))
+      return r;
+
+    if((uint64_t)time(NULL) <= req.timestamp)
+    {
+      error_resp.code = CORE_RPC_ERROR_CODE_TOO_BIG_TIMESTAMP;
+      error_resp.message = "Requested timestamp is in the future";
+      return false;
+    }
+    CHECK_PAYMENT(req, res, COST_PER_HEIGHT_BY_TIME);
+
+    constexpr uint64_t buffer_time = 1 * 60 * 60; //1 hour
+    constexpr uint64_t buffer_height = 1 * 30;    //1 hour
```

Comment:
Is this 30 from 60 minutes in an hour divided by a block every two minutes? In other words, what is `buffer_height`?

Reply:
Refer to lines 3638-3642. The binary search ends when the boundary block heights are closer than `buffer_height`. The search uses both a time-based and a height-based condition to ensure a good behavior even with irregular block times.